docs/source/specs/openapi.json

```diff
@@ -1434,6 +1434,46 @@
           }
         ]
       },
+      "GroupWithPrincipalsAndRoles": {
+        "allOf": [{
+            "$ref": "#/components/schemas/Group"
+          },
+          {
+            "$ref": "#/components/schemas/UUID"
+          },
+          {
+            "$ref": "#/components/schemas/Timestamped"
+          },
+          {
+            "type": "object",
+            "required": [
+              "principals"
+            ],
+            "properties": {
+              "principals": {
+                "type": "array",
+                "items": {
+                  "$ref": "#/components/schemas/Principal"
+                }
+              }
+            }
+          },
+          {
+            "type": "object",
```

Comment:
Any reason why this is two `type: object` references?
I would have done the yaml as:
```
    GroupWithPrincipalsAndRoles:
      allOf:
        - $ref: '#/components/schemas/Group'
        - $ref: '#/components/schemas/UUID'
        - $ref: '#/components/schemas/Timestamped'
        - type: object
          required:
            - principals
            - roles
          properties:
            principals:
              type: array
              items:
                $ref: '#/components/schemas/Principal'
            roles:
              type: array
              items:
                $ref: '#/components/schemas/Role'
```

With the resulting JSON:
```
      "GroupWithPrincipalsAndRoles": {
        "allOf": [
          {
            "$ref": "#/components/schemas/Group"
          },
          {
            "$ref": "#/components/schemas/UUID"
          },
          {
            "$ref": "#/components/schemas/Timestamped"
          },
          {
            "type": "object",
            "required": [
              "principals",
              "roles"
            ],
            "properties": {
              "principals": {
                "type": "array",
                "items": {
                  "$ref": "#/components/schemas/Principal"
                }
              },
              "roles": {
                "type": "array",
                "items": {
                  "$ref": "#/components/schemas/Role"
                }
              }
            }
          }
        ]
      },
```

Doesn't seem to complain in the editor, so it may not be wrong to do it that way. I was just wondering.

Reply:
No, I think you're right. I've updated it. Thanks!